crypto/merkletrie/trie.go

```diff
@@ -63,7 +63,7 @@ type Trie struct {
 	root                storedNodeIdentifier
 	nextNodeID          storedNodeIdentifier
 	lastCommittedNodeID storedNodeIdentifier
-	cache               *merkleTrieCache
+	cache               merkleTrieCache
```

Comment:
question: most of the changes in the PR that changes from `mt.cache` to `&mt.cache` are related with this line.

Is it because of memory management that we want to change from pointer to instance of cache?

Reply:
Yeah, it just didn't need to be a pointer, so I figured we'd avoid the allocation.